src/cargo/ops/cargo_fetch.rs

```diff
@@ -57,6 +58,34 @@ pub fn fetch<'a>(
             .map(|(id, _deps)| id);
         deps_to_fetch.extend(deps);
     }
+
+    // If -Zbuild-std was passed, download dependencies for the standard library.
+    if let Some(crates) = &config.cli_unstable().build_std {
+        if build_config.build_plan {
+            config
+                .shell()
+                .warn("-Zbuild-std does not currently fully support --build-plan")?;
+        }
+        if build_config.requested_kinds[0].is_host() {
+            // TODO: This should eventually be fixed. Unfortunately it is not
+            // easy to get the host triple in BuildConfig. Consider changing
+            // requested_target to an enum, or some other approach.
+            anyhow::bail!("-Zbuild-std requires --target");
+        }
+        let target_data = RustcTargetData::new(ws, &build_config.requested_kinds)?;
+        let (std_package_set, std_resolve, _) =
+            standard_lib::resolve_std(ws, &target_data, &build_config.requested_kinds, &crates)?;
```

Comment:
This is mostly copied from `create_bcx` in `cargo_compile.rs`, hopefully that's ok.